src/containers/ConversionForm/index.tsx

```diff
@@ -55,15 +62,46 @@ const ConversionForm = ({
     const [fileToConvert, setFileToConvert] = useState<UploadFile>();
     const [engineSelected, setEngineSelected] = useState<boolean>(false);
     const [isProcessing, setIsProcessing] = useState<boolean>(false);
+    const [fileTypeErrorModalOpen, setFileTypeErrorModalOpen] = useState(false);
 
+    const toggleModal = () => {
+        setFileTypeErrorModalOpen(!fileTypeErrorModalOpen);
+    };
     const toggleProcessing = () => {
         setIsProcessing(!isProcessing);
     };
 
+    const handleEngineChange = (selectedValue: string) => {
+        const selectedEngine = selectedValue as AvailableEngines;
+        setConversionEngine(selectedEngine);
+        setEngineSelected(true);
+    };
+
+    const validateFileType = () => {
+        if (fileToConvert) {
+            const fileExtension =
+                fileToConvert.name.split(".").pop()?.toLowerCase() || "";
+            if (
+                validFileExtensions[conversionProcessingData.engineType] ===
+                fileExtension
+            ) {
+                setIsProcessing(!isProcessing);
+                return;
+            }
+        }
+        setFileTypeErrorModalOpen(true);
+    };
+
```

Comment:
New handler sets engine in store, and validateFileType checks extension against that. More robust this way and no new state needed.